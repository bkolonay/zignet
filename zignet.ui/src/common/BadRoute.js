import React, { Component } from 'react';

class BadRoute extends Component {

  render() {
    return (
      <p>Route does not exist</p>
    );
  }
}

export default BadRoute;
