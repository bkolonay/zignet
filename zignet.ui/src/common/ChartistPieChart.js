import React, { Component } from 'react';
import Chartist from 'chartist';
import './chartist.css';

class ChartistPieChart extends Component {

  componentDidMount() {
  	this.chartistChart = Chartist.Pie(
  		this.chartContainer,
  		this.props.chartData,
  		{ 
  			donut: true,
			donutWidth: 40
		}
	)
  }

  render() {
    return (
      <div 
        className="ct-chart" 
        id={"chart" + this.props.chartId}
        ref={(div) => this.chartContainer = div}
      />
    );
  }
}

export default ChartistPieChart;
