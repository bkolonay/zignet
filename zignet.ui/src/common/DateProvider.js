import moment from 'moment';

function now() {
	return moment();
}

export { now }