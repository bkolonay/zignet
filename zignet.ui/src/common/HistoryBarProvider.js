import moment from 'moment';

class HistoryBarProvider {

  /* 
    helped with percentage calculations:
     - https://www.calculatorsoup.com/calculators/math/percentage.php
     - https://www.timeanddate.com/date/timeduration.html
  */
  
	getErrorDivAttributes(historyBarWidth, now, testFailureDuration) {
		let oneDayAgo = moment(now.format()).subtract(24, 'hours');
    let testFailureStartTime = moment.utc(testFailureDuration.FailureStart).local();
    let minutesBetweenOneDayAgoAndTestFailureStartTime = moment.duration(testFailureStartTime.diff(oneDayAgo)).asMinutes();

    var failureDivStart = 0;
    if (minutesBetweenOneDayAgoAndTestFailureStartTime > 0) {
      let passingPercentOfLastDay = (minutesBetweenOneDayAgoAndTestFailureStartTime / 1440) * 100;
      failureDivStart = Math.floor((passingPercentOfLastDay / 100) * historyBarWidth);
    }
    else
      testFailureStartTime = oneDayAgo;

    var failureDivWidth = 0;
    if (testFailureDuration.FailureEnd) {
      let testFailureEndTime = moment.utc(testFailureDuration.FailureEnd).local();
      let failureDurationInMinutes = moment.duration(testFailureEndTime.diff(testFailureStartTime)).asMinutes();
      let failingPercentOfLastDay = (failureDurationInMinutes / 1440) * 100;
      failureDivWidth = Math.floor((failingPercentOfLastDay / 100) * historyBarWidth);
      if (failureDivWidth < 1)
      	failureDivWidth = 1;
    }
    else 
      failureDivWidth = historyBarWidth - failureDivStart;

    return {left: failureDivStart, width: failureDivWidth};
	}

  getDivTitle(testFailureDuration) {
    let testFailureStartTime = moment.utc(testFailureDuration.FailureStart).local();
    if (testFailureDuration.FailureEnd)
    {
      let testFailureEndTime = moment.utc(testFailureDuration.FailureEnd).local();
      let timeFromStartToEnd = testFailureStartTime.to(testFailureEndTime, true);
      return timeFromStartToEnd + ' from ' + testFailureStartTime.format('l h:mma') + ' - ' + testFailureEndTime.format('l h:mma');
    }
    else
    {
      let timeFromStartToEnd = testFailureStartTime.toNow(true);
      return timeFromStartToEnd + ' from ' + testFailureStartTime.format('l h:mma') + ' - now';
    }      
  }
}

export default HistoryBarProvider;