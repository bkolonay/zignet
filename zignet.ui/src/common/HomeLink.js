import React, { Component } from 'react';
import { getUrlParams } from '../common/routing/SearchFilter.js'

class HomeLink extends Component {

  render() {
    return (
      <a href={"/?" + getUrlParams(this.props.filter)}>ZigNet</a>
    );
  }
}

export default HomeLink;
