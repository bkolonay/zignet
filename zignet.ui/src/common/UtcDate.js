import moment from 'moment';

class UtcDate {
  constructor(date) {
    this.moment = moment.utc(date);
  }

  toLocal() {
    this.moment = this.moment.local();
    return this;
  }

  getDateAndTimeString() {
    return this.moment.format('L LTS');
  }

  getTimeFromNow() {
    return this.moment.fromNow(true);
  }

  getTimeFromNowWithSuffix() {
    return this.moment.fromNow();
  }  

  getTimeFrom(date) {
    const localMoment = moment.utc(date);
    return this.moment.from(localMoment, true);
  }  
}

export default UtcDate;
