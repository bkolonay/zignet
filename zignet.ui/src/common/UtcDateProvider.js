import moment from 'moment';

function getTimeFromNow(utcDateTime) {
	let localMoment = moment.utc(utcDateTime);
	return localMoment.fromNow(true);
}

function getTimeFromNowWithSuffix(utcDateTime) {
	let localMoment = moment.utc(utcDateTime);
	return localMoment.fromNow();
}

export { getTimeFromNow, getTimeFromNowWithSuffix }