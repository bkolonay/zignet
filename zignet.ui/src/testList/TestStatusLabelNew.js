import React, { Component } from 'react';
import { getTimeFromNow } from '../common/UtcDateProvider'

class TestStatusLabelNew extends Component {

  render() {
    const label = this.props.failingFromDate ? 
      <span className='text-danger'>{'for ' + getTimeFromNow(this.props.failingFromDate)}</span> :
      <span className='text-success'>{'for ' + getTimeFromNow(this.props.passingFromDate)}</span>;

    return (
      label
    );
  }
}

export default TestStatusLabelNew;
