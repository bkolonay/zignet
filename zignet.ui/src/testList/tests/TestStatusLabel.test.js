import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestStatusLabelNew from '../TestStatusLabelNew';
import { getTimeFromNow } from '../../common/UtcDateProvider';

jest.mock('../../common/UtcDateProvider');

it('renders when has failing from date', () => {
  const component = <TestStatusLabelNew failingFromDate={'2018-05-24T01:00:00'} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('renders when does not have failing from date', () => {
  const component = <TestStatusLabelNew failingFromDate={undefined} passingFromDate={'2018-06-05T01:00:00'} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('snapshot when has failing from date', () => {
  getTimeFromNow.mockReturnValue('10 minutes');
  const component = <TestStatusLabelNew failingFromDate={'2018-05-24T01:00:00'} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when does not have failing from date', () => {
  getTimeFromNow.mockReturnValue('4 days');
  const component = <TestStatusLabelNew failingFromDate={undefined} passingFromDate={'2018-06-05T01:00:00'} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});