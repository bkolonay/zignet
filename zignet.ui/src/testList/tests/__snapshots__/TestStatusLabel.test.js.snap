// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot when does not have failing from date 1`] = `
<span
  className="text-success"
>
  for 4 days
</span>
`;

exports[`snapshot when has failing from date 1`] = `
<span
  className="text-danger"
>
  for 10 minutes
</span>
`;
